projects/libs/core-library-typescript/test/consensus/block-builder.spec.ts

```diff
@@ -160,16 +107,20 @@ describe("a block", () => {
     });
 
     it("is built from pending transactions shortly after started", (done) => {
-      transactionPool.getAllPendingTransactions.returns(pullAllPendingTransactionsOutput);
-      virtualMachine.processTransactionSet.returns(processTransactionSetOutput);
-
       blockBuilder.start();
 
       setTimeout(() => {
-        expect(newBlockBuildCallback).to.have.been.calledWith(sinon.match.has("body",
-        sinon.match.has("transactions", dummyTransactionSet).and(sinon.match.has("stateDiff", dummyStateDiff))));
-        done();
-        blockBuilder.stop();
+        try {
+          const bodyMatch = sinon.match.has("transactions", dummyTransactionSet)
+          .and(sinon.match.has("stateDiff", dummyStateDiff))
+          .and(sinon.match.has("transactionReceipts"));
```

Comment:
I think maybe there is a need for more extensive tests that validates contents of `transactionReceipts` including transactions that were rejected or failed.

Reply:
👍

Reply:
Actually, this should not be in the scope of the Block Builder test. The transaction receipts are propagated from the virtual machine and assumed to be generated correctly

Reply:
Do we have a test like this in virtual machine?